bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ImageImpl.java

```diff
@@ -238,6 +244,32 @@ protected void initModel() {
         }
     }
 
+    protected String getAssetName() {
+        if (StringUtils.isNotBlank(fileReference)) {
+            Resource damResource = request.getResourceResolver().getResource(fileReference);
+            if (damResource != null) {
+              Asset asset = damResource.adaptTo(Asset.class);
+              //check if the image name is overwritten at the dialog level, else set the DAM asset name
+              final String assetName = StringUtils.isNotBlank(imageName) ? imageName : (asset != null ? asset.getName() : "");
```

Comment:
Slight performance improvement: Move the check for a set image name to the beginning of the method, then you don't need to get the underlying asset at all!

Reply:
thanks for this, I have added this check even one level before now.